test/test_swa.py

```diff
@@ -0,0 +1,692 @@
+            adadelta = optim.Adadelta(params)
+            return contriboptim.SWA(
+                adadelta, swa_start=1000, swa_freq=1)
+
+        def adagrad_constructor(params):
+            adagrad = optim.Adagrad(params, lr=1e-1)
+            return contriboptim.SWA(
+                adagrad, swa_start=1000, swa_freq=1, swa_lr=1e-2)
+
+        def adamax_constructor(params):
+            adamax = optim.Adamax(params, lr=1e-1)
+            return contriboptim.SWA(
+                adamax, swa_start=1000, swa_freq=1, swa_lr=1e-2)
+
+        def rmsprop_constructor(params):
+            rmsprop = optim.RMSprop(params, lr=1e-2)
+            return contriboptim.SWA(
+                rmsprop, swa_start=1000, swa_freq=1, swa_lr=1e-3)
+
+        def rprop_constructor(params):
+            rprop = optim.Rprop(params, lr=1e-2)
+            return contriboptim.SWA(
+                rprop, swa_start=1000, swa_freq=1, swa_lr=1e-3)
+
+        def asgd_constructor(params):
+            asgd = optim.ASGD(params, lr=1e-3)
+            return contriboptim.SWA(
+                asgd, swa_start=1000, swa_freq=1, swa_lr=1e-3)
+
+        def lbfgs_constructor(params):
+            lbfgs = optim.LBFGS(params, lr=5e-2, max_iter=5)
+            return contriboptim.SWA(
+                lbfgs, swa_start=1000, swa_freq=1, swa_lr=1e-3)
+
+        auto_constructor_list = [sgd_constructor, sgd_momentum_constructor, 
+                                 adam_constructor, adadelta_constructor, 
+                                 adagrad_constructor, adamax_constructor, 
+                                 rmsprop_constructor, rprop_constructor,
+                                 asgd_constructor, lbfgs_constructor]
+
+        #auto_constructor_list = [sgd_momentum_constructor]
+
+        for i, constructor in enumerate(auto_constructor_list):
+            print(constructor)
+            # Pass
+            self._test_rosenbrock(constructor)
+            self._test_basic_cases(
+                    lambda weight, bias: constructor([weight, bias]))
+            if i < len(auto_constructor_list) - 1:
+                self._test_basic_cases(
+                        lambda weight, bias: constructor(
+                            self._build_params_dict(weight, bias, lr=1e-2)))
+                self._test_basic_cases(
+                        lambda weight, bias: constructor(
+                            self._build_params_dict_single(weight, bias, lr=1e-2)))
+
+
+        self._test_rosenbrock(sgd_manual_constructor, automode=False)
+
+    def _define_vars_loss_opt(self):
+        x = Variable(torch.Tensor([5., 2.]), requires_grad=True)
+        y = Variable(torch.Tensor([3., 7.]), requires_grad=True)
+
+        def loss_fun(a, b):
+            return torch.sum(a * b)**2
+
+        opt = optim.SGD([{'params': [x]},
+                        {'params': [y], 'lr': 1e-3}], lr=1e-2, momentum=0.9)
+        return x, y, loss_fun, opt
+
+    def testSWAAuto(self):
```

Comment:
Please use `snake_case` just like in the rest of the code base

Reply:
done